app/code/BVMyBlog/Blog/Block/LastPosts.php

```diff
@@ -0,0 +1,115 @@
+<?php
+declare(strict_types=1);
+
+namespace BVMyBlog\Blog\Block;
+
+use BVMyBlog\Blog\Model\BlogRepository;
+use Magento\Catalog\Model\Locator\RegistryLocator;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\SortOrderBuilder;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Exception\NotFoundException;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+
+/**
+ * Returns 5 latest posts
+ */
+class LastPosts extends Template
+{
+    /**
+     * @var SortOrderBuilder $sortOrderBuilder
+     */
+    private $sortOrderBuilder;
+
+    /**
+     * @var SearchCriteriaBuilder $searchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var BlogRepository $blogRepository
+     */
+    private $blogRepository;
+
+    /**
+     * ScopeConfigInterface $scopeInterface
+     */
+    private $scopeInterface;
+
+    /**
+     * @var RegistryLocator $registryLocator
+     */
+    private $registryLocator;
+
+    /**
+     * @param ScopeConfigInterface $scopeInterface
+     * @param RegistryLocator $registryLocator
+     * @param SortOrderBuilder $sortOrderBuilder
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param BlogRepository $blogRepository
+     * @param Context $context
+     * @param array $data
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeInterface,
+        RegistryLocator $registryLocator,
+        SortOrderBuilder $sortOrderBuilder,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        BlogRepository $blogRepository,
+        Context $context,
```

Comment:
When extending some class, constructor arguments should be in next order
- Parent Required Arguments
- Child Required Arguments
- Optional Arguments